server/handlers.go

```diff
@@ -993,6 +1000,131 @@ func handleCommunityTokenImages(db *sql.DB, logger *zap.Logger) http.HandlerFunc
+		if err != nil {
+			return
+		}
+		if communityDescription.Identity == nil {
+			logger.Error("no identity in community description", zap.String("community id", communityID))
+			return
+		}
+
+		var imagePayload []byte
+		for t, i := range communityDescription.Identity.Images {
+			if t == name {
+				imagePayload = i.Payload
+			}
+		}
+		if imagePayload == nil {
+			logger.Error("can't find community description image", zap.String("community id", communityID), zap.String("name", name))
+			return
+		}
+
+		mime, err := images.GetProtobufImageMime(imagePayload)
+		if err != nil {
+			logger.Error("failed to get community image mime", zap.String("community id", communityID), zap.Error(err))
+		}
+
+		w.Header().Set("Content-Type", mime)
+		w.Header().Set("Cache-Control", "no-store")
+		_, err = w.Write(imagePayload)
+		if err != nil {
+			logger.Error("failed to write community image", zap.String("community id", communityID), zap.Error(err))
+		}
+	}
+}
+
+func handleCommunityDescriptionTokenImagesPath(db *sql.DB, logger *zap.Logger) http.HandlerFunc {
+	if db == nil {
+		return handleRequestDBMissing(logger)
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		params := r.URL.Query()
+
+		if len(params["communityID"]) == 0 {
+			logger.Error("[handleCommunityDescriptionTokenImagesPath] no communityID")
+			return
+		}
+		communityID := params["communityID"][0]
+
+		if len(params["symbol"]) == 0 {
+			logger.Error("[handleCommunityDescriptionTokenImagesPath] no symbol")
+			return
+		}
+		symbol := params["symbol"][0]
+
+		err, communityDescription := getCommunityDescription(db, communityID, logger)
+		if err != nil {
+			return
+		}
+
+		var foundToken *protobuf.CommunityTokenMetadata
+		for _, m := range communityDescription.CommunityTokensMetadata {
+			if m.GetSymbol() == symbol {
+				foundToken = m
+			}
+		}
+		if foundToken == nil {
+			logger.Error("can't find community description token image", zap.String("community id", communityID), zap.String("symbol", symbol))
+			return
+		}
+
+		imagePayload, err := images.GetPayloadFromURI(foundToken.Image)
+		if err != nil {
+			logger.Error("failed to get community description token image payload", zap.Error(err))
+			return
+		}
+		mime, err := images.GetProtobufImageMime(imagePayload)
+		if err != nil {
+			logger.Error("failed to get community description token image mime", zap.String("community id", communityID), zap.String("symbol", symbol), zap.Error(err))
+		}
+
+		w.Header().Set("Content-Type", mime)
+		w.Header().Set("Cache-Control", "no-store")
+		_, err = w.Write(imagePayload)
+		if err != nil {
+			logger.Error("failed to write community description token image", zap.String("community id", communityID), zap.String("symbol", symbol), zap.Error(err))
+		}
+	}
+}
+
+// getCommunityDescription returns the latest community description from the cache.
+// NOTE: you should ensure preprocessDescription is called before this function.
```

Comment:
Sorry if this is obvious, but when exactly should `preprocessDescription` be called?

The flow that I understand is:

1. Client calls `/communityDescriptionImages`
2. Function `getCommunityDescription` is called inside `handleCommunityDescriptionImagesPath`.

So there's no control at the media server level for calling `preprocessDescription`. How can this recommendation be followed?

Reply:
Got it, thanks!